devops/azure-pipelines/maven-build-test-job-template.yml

```diff
@@ -0,0 +1,50 @@
+parameters:
+  - name: 'jobName'
+    type: string
+  - name: 'serviceConnection'
+    type: string
+  - name: 'organization'
+    type: string
+  - name: 'testPath' # file path to test results
+    type: string
+jobs:
+  - job: ${{ parameters.jobName }}
+    steps:
+      - task: SonarCloudPrepare@1
+        condition: ne(variables['Build.Reason'], 'PullRequest')
+        inputs:
+          SonarCloud: 'Forge-SonarCloud'
+          organization: '1058-sophia-gavrila-forge'
+          scannerMode: 'Other'
+          extraProperties: |
+           sonar.branch.name=$(Build.SourceBranchName) 
```

Comment:
not needed; sonar will auto-detect it from the .git folder. also, fix the indent on this block

Reply:
Well, we're getting a "Error: could not find a default branch to fall back on" without the "sonar.branch.name", and supplying the sonar.branch.name was one way we got that error to go away. Googling the error gets stuff about "check your project/organization key" as well as "[check your default branch](https://community.sonarsource.com/t/could-not-find-a-default-branch-to-fall-back-on/25095)", and if memory serves me right pretty sure that was the same error text when we learned about SonarQube not being happy with PRs.

Reply:
Well, I just re-ran the pipeline on a different branch where I removed the sonar branch name, and it worked. Maybe the problem was that the could not find default branch to fall back on was when we hadn't yet even run the sonar scan for the main branch, but when we have seeded it with a main branch, we no longer need it... (not sure if this is right, but it could explain the symptoms)

Reply:
that may be right.